addons/l10n_id_efaktur/models/efaktur.py

```diff
@@ -14,44 +14,55 @@ class Efaktur(models.Model):
     company_id = fields.Many2one('res.company', required=True, default=lambda self: self.env.company)
     max = fields.Char(compute='_compute_default', store=True, readonly=False)
     min = fields.Char(compute='_compute_default', store=True, readonly=False)
-    available = fields.Integer(compute='_compute_available', store=True)
+    available = fields.Integer(readonly=True)
+    next_num = fields.Integer(compute="_compute_next_num", store=True)
+
+    @api.depends('max', 'available')
+    def _compute_next_num(self):
+        for record in self:
+            record.next_num = int(record.max) - record.available + 1
 
     @api.model
     def pop_number(self, company_id):
-        range = self.search([('company_id', '=', company_id)], order="min ASC", limit=1)
-        if not range:
+        erange = self.search([('company_id', '=', company_id)], order="next_num ASC", limit=1)
+
+        if not erange:
             return None
 
-        popped = int(range.min)
-        if int(range.min) >= int(range.max):
-            range.unlink()
-        else:
-            range.min = '%013d' % (popped + 1)
+        popped = erange.next_num
+        erange.available -= 1
+
         return popped
 
     @api.model
     def push_number(self, company_id, number):
-        return self.push_numbers(company_id, number, number)
-
-    @api.model
-    def push_numbers(self, company_id, min, max):
-        range_sup = self.search([('min', '=', '%013d' % (int(max) + 1))])
-        if range_sup:
-            range_sup.min = '%013d' % int(min)
-            max = range_sup.max
-
-        range_low = self.search([('max', '=', '%013d' % (int(max) - 1))])
-        if range_low:
-            range_sup.unlink()
-            range_low.max = '%013d' % int(max)
-
-        if not range_sup and not range_low:
-            self.create({
-                'company_id': company_id,
-                'max': '%013d' % int(max),
-                'min': '%013d' % int(min),
-            })
-
+        # if released number is the last released number from a range, extend the availability
+        erange = self.search([('company_id', '=', company_id), ('next_num', '=', int(number) + 1)])
```

Comment:
Here you want to search explicitely for the range in which number is between the min and max like, below, you can check if it's the last released of a range or no afterward